cli/lib/gateway.js

```diff
@@ -337,3 +340,49 @@ function hasConfigChanged(oldConfig, newConfig) {
         return true;
     }
 }
+
+function validator(newConfig) {
+    
+    //checkObject(newConfig.product_to_proxy) && 
+    //checkObject(newConfig.product_to_api_resource)
+
+    if (checkObject(newConfig) &&
+        checkObject(newConfig.analytics) && 
+        checkObject(newConfig.analytics.source) && 
+        checkObject(newConfig.analytics.proxy) && 
+        checkObject(newConfig.analytics.key) && 
+        checkObject(newConfig.analytics.secret) &&
+        checkObject(newConfig.analytics.uri) &&
+        checkObject(newConfig.edgemicro) && 
+        checkObject(newConfig.edgemicro.port) && 
+        checkObject(newConfig.edgemicro.max_connections) &&
+        checkObject(newConfig.headers) && 
+        checkArray(newConfig.proxies)) { 
+        debug("configuration incomplete or invalid, skipping configuration");
+        return false;
+    }
+
+    return true;
+}
+
+function checkObject (o) {
+    if (o === null) {
+        return false;
+    }
+    if (typeof o === 'object' && o instanceof Object && !(o instanceof Array)) {
```

Comment:
can reduce number of lines by omitting if else to `return (typeof o === 'object' && o instanceof Object && !(o instanceof Array))`